h2o-core/src/main/java/water/init/AbstractEmbeddedH2OConfig.java

```diff
@@ -38,16 +38,31 @@
    */
   public abstract String fetchFlatfile() throws Exception;
 
+  /**
+   * @deprecated please override version of this method that has leader information.
+   */
+  @Deprecated
+  public void notifyAboutCloudSize(InetAddress ip, int port, int size) {
+    throw new IllegalStateException("Please override method new version of the method notifyAboutCloudSize(..).");
+  }
+
   /**
    * Tell the embedding software that this H2O instance belongs to
    * a cloud of a certain size.
    * This may be nonblocking.
    *
+   * Note: this method will be made abstract in a future stable release. User is expected to override the method
+   * in the implementation.
+   *
    * @param ip IP address this H2O can be reached at.
    * @param port Port this H2O can be reached at (for REST API and browser).
+   * @param leaderIp IP address of the leader H2O node of the cloud.
+   * @param leaderPort Port of the leader H2O node (for REST API and browser).
```

Comment:
What if there is no REST API exposed?

Reply:
Hmm, a case when the leader doesn't have REST API exposed but the current member does?

Such clusters cannot be created from the  hadoop driver but they are possible in general. Do we need to consider this case?

Reply:
That's case of SW - we are using EmbeddedH2OConfig, but running each executor h2o with `--disable_web`.

Reply:
But that is still okay (or at least consistent with current behavior).

Currently we report embeddedWebServerIp/Port regardless if the node has enabled REST or it doesn't.

Reply:
oki, so no harm if we send bogus port

Reply:
Yes, I think that is the case.

There will only be an issue if the cluster was not homogeneous (some non-client nodes started with `--disable_web` and some not). Although this is theoretically possible I don't think we need to be concerned with the use case.